src/napari_matplotlib/base.py

```diff
@@ -69,25 +73,55 @@ def add_single_axes(self) -> None:
         self.axes = self.figure.subplots()
         self.apply_napari_colorscheme(self.axes)
 
-    @staticmethod
-    def apply_napari_colorscheme(ax: Axes) -> None:
+    def apply_napari_colorscheme(self, ax: Axes) -> None:
         """Apply napari-compatible colorscheme to an Axes."""
+        # get the foreground colours from current theme
+        theme = napari.utils.theme.get_theme(self.viewer.theme, as_dict=False)
+        fg = theme.foreground.as_hex()  # fg is a muted contrast to bg
+        tx = theme.text.as_hex()  # text is high contrast to bg
+
         # changing color of axes background to transparent
         ax.set_facecolor("none")
 
         # changing colors of all axes
         for spine in ax.spines:
-            ax.spines[spine].set_color("white")
+            ax.spines[spine].set_color(fg)
 
-        ax.xaxis.label.set_color("white")
-        ax.yaxis.label.set_color("white")
+        ax.xaxis.label.set_color(tx)
+        ax.yaxis.label.set_color(tx)
 
         # changing colors of axes labels
-        ax.tick_params(axis="x", colors="white")
-        ax.tick_params(axis="y", colors="white")
+        ax.tick_params(axis="x", colors=tx)
+        ax.tick_params(axis="y", colors=tx)
+
+    def _on_theme_change(self) -> None:
+        """Update MPL toolbar and axis styling when `napari.Viewer.theme` is changed.
+
+        Note:
+            At the moment we only handle the default 'light' and 'dark' napari themes.
+        """
+        self._replace_toolbar_icons()
+        if self.figure.gca():
+            self.apply_napari_colorscheme(self.figure.gca())
+
+    def _get_path_to_icon(self) -> Path:
+        """
+        Get the icons directory (which is theme-dependent).
+        """
+        # TODO: can make this more robust by doing some RGB tricks to figure out
+        # whether white or black icons are going to be more visible given the
+        # theme.background
+        islight = self.viewer.theme == "light"
+        if islight:
+            return ICON_ROOT / "black"
+        else:
+            return ICON_ROOT / "white"
```

Comment:
Oh, forgot to mention in my blah blah, `ICON_ROOT` doesn't have to be global anymore.